iree/hal/vulkan/emulated_timeline_semaphore.cc

```diff
@@ -200,24 +204,42 @@ StatusOr<VkSemaphore> EmulatedTimelineSemaphore::GetSignalSemaphore(
 
 StatusOr<bool> EmulatedTimelineSemaphore::TryToAdvanceTimeline(
     uint64_t to_upper_value) {
-  IREE_TRACE_SCOPE0("EmulatedTimelineSemaphore::TryToAdvanceTimeline");
+  absl::InlinedVector<VkFence, 4> signaled_fences;
+  auto status = TryToAdvanceTimeline(to_upper_value, &signaled_fences);
```

Comment:
Why defer checking the status until after calling `on_fence_signal_`?

Reply:
I don't think it matters that much? If errors happen, either returning immediately or still sending the queue a notice to clean up something are both okay to me..